dlib/test/vectorstream.cpp

```diff
@@ -71,13 +72,42 @@ namespace
 
         DLIB_TEST(s.peek() == EOF);
         DLIB_TEST(s.get() == EOF);
+        
+        s.clear();
+        s.seekg(1000, std::ios_base::beg); //read_pos should be 1000
```

Comment:
Add a test right after this line to check that the stream isn't in a funny state.  E.g. you should be able to read some values and see they are what is expected.